### ISSUE TITLE
Add "@context" as part of the special attributes of an entity

### ISSUE THREAD

**Issue by jmcanterafonseca** [telefonicaid/fiware-orion#2993]
In order to smooth the path towards JSON-LD support, it would be good to add "@context" as one of the special attributes of an entity, together with "type" and "id". It would be an optional "property" of an entity. 

If present, "@context", it will always have to be rendered whenever "id" and "type" are rendered. 

See "@context" specification at https://www.w3.org/TR/2014/REC-json-ld-20140116/#the-context

**Comment by jmcanterafonseca** [comment]
@McMutton would you like to work on this issue? at least start analysing it?

**Comment by fgalan** [comment]
I understand this @context field may be useful/needed for the ETSI-CIM version of the API but, given the work is yet in progress (until end of 2017?). Thus I don't recommend to work on it now.

**Comment by jmcanterafonseca** [comment]
@fgalan I have updated the issue after a double check. I think it is a feature we need to pave the way towards ETSI CIM and at least it would deserve analysis at this stage

**Comment by fgalan** [comment]
Not sure if I'm getting the point...

> It [the `@context`] would be an optional "property" of an entity, provided at creation time or modified later.

Actually, current Orion allows you to create such property as a regular attribute:

```
curl localhost:1026/v2/entities -s -S --header 'Content-Type: application/json' -d @- <<EOF
{
  "id": "Room1",
  "type": "Room",
  "@context": {
    "type": "Text",
    "value": "My marvellous context"
  },
  "temperature": {
    "value": 23,
    "type": "Float"
  }
}
EOF
```

and retrieve/modify using regular NGSIv2:

```
curl -s -S  localhost:1026/v2/entities/Room1 | python -mjson.tool
{
    "@context": {
        "metadata": {},
        "type": "Text",
        "value": "My marvellous context"
    },
    "id": "Room1",
    "temperature": {
        "metadata": {},
        "type": "Float",
        "value": 23
    },
    "type": "Room"
}
```

So in some way (and at least the way we need "to pave the way") it is already implemented and I confirm my recommendation on not working more on this line at the present moment. Once ETSI-CIM gets release, we'll see.

**Comment by jmcanterafonseca** [comment]
yes, but the problem is that we do not want to be caught in the trap of {
type, value, metadata } of a regular NGSIv2 attribute. We want this
attribute to be a special one and to allow it to be just an array or an
object as the JSON-LD states.

2017-10-04 13:56 GMT+02:00 Fermín Galán Márquez <notifications@github.com>:
